python/lsst/pipe/tasks/hips.py

```diff
@@ -604,3 +614,907 @@ def build_quantum_graph(
+        initial_dec : `float`
+            Initial HiPS Dec position (degrees).
+        initial_fov : `float`
+            Initial HiPS display size (degrees).
+        """
+        # WARNING: In general PipelineTasks are not allowed to do any outputs
+        # outside of the butler.  This task has been given (temporary)
+        # Special Dispensation because of the nature of HiPS outputs until
+        # a more controlled solution can be found.
+        def _write_property(fh, name, value):
+            """Write a property name/value to a file handle.
+
+            Parameters
+            ----------
+            fh : file handle (blah)
+                Open for writing.
+            name : `str`
+                Name of property
+            value : `str`
+                Value of property
+            """
+            for char in name:
+                if char.isspace():
+                    raise ValueError(f"``{name}`` cannot contain any space characters.")
+            if '=' in name:
+                raise ValueError(f"``{name}`` cannot contain an ``=``")
+
+            fh.write(f"{name:25}= {value}\n")
+
+        if exposure.image.array.dtype == np.dtype('float32'):
+            bitpix = -32
+        elif exposure.image.array.dtype == np.dtype('float64'):
+            bitpix = -64
+        elif exposure.image.array.dtype == np.dtype('int32'):
+            bitpix = 32
+
+        date_iso8601 = datetime.utcnow().isoformat(timespec='seconds') + 'Z'
+        pixel_scale = np.rad2deg(hp.nside2resol(2**(max_order + shift_order)))
+
+        uri = hips_base_path.join("properties")
+        with ResourcePath.temporary_uri(suffix=uri.getExtension()) as temporary_uri:
+            with open(temporary_uri.ospath, "w") as fh:
+                _write_property(
+                    fh,
+                    "creator_did",
+                    property_config.creator_did_template.format(band=band),
+                )
+                if property_config.obs_collection is not None:
+                    _write_property(fh, "obs_collection", property_config.obs_collection)
+                _write_property(
+                    fh,
+                    "obs_title",
+                    property_config.obs_title_template.format(band=band),
+                )
+                if property_config.obs_description_template is not None:
+                    _write_property(
+                        fh,
+                        "obs_description",
+                        property_config.obs_description_template.format(band=band),
+                    )
+                if len(property_config.prov_progenitor) > 0:
+                    for prov_progenitor in property_config.prov_progenitor:
+                        _write_property(fh, "prov_progenitor", prov_progenitor)
+                if property_config.obs_ack is not None:
+                    _write_property(fh, "obs_ack", property_config.obs_ack)
+                _write_property(fh, "obs_regime", "Optical")
```

Comment:
Would this chunk of code look easier to read if you constructed a dict and then had `_write_property_dict()` ? Or did `for k, v in property_dict: _write_property(fh, k, v)`? That could also allow the dict to be constructed outside of the `with` block.

Reply:
The `prov_progenitor` property in particular is allowed to have multiple entries (and there are others) so that makes handling the dict more complicated.  I guess it could have a value that is a str or a list, but then the code simplification gain is much smaller.